### ISSUE TITLE
fstadv_forecasts fails for EP031999

### ISSUE THREAD

**Issue by timtrice** [ropensci/rrricanes#53]
Error:

```
Error: `.x` (0) and `.y` (2) are different lengths

In addition: Warning message:
All formats failed to parse. No formats found. 
```

Traceback:

```
19.
purrr::map2(., 1:n, function(a, b) {
    if (tmp[[b]]$wf != "") {
        wf <- fstadv_forecast_wind_radius(tmp[[b]]$wf)
        tmp[[b]] <- dplyr::bind_cols(tmp[[b]], wf) ... 
18.
function_list[[k]](value) 
17.
withVisible(function_list[[k]](value)) 
16.
freduce(value, `_function_list`) 
15.
`_fseq`(`_lhs`) 
14.
eval(quote(`_fseq`(`_lhs`)), env, env) 
13.
eval(quote(`_fseq`(`_lhs`)), env, env) 
12.
withVisible(eval(quote(`_fseq`(`_lhs`)), env, env)) 
11.
tmp %>% purrr::map2(1:n, function(a, b) {
    if (tmp[[b]]$wf != "") {
        wf <- fstadv_forecast_wind_radius(tmp[[b]]$wf)
        tmp[[b]] <- dplyr::bind_cols(tmp[[b]], wf) ... at fstadv.R#390
10.
fstadv_forecasts(contents, date) at fstadv.R#249
9.
.f(.x[[i]], ...) 
8.
purrr::map(filter_fstadv(products), fstadv) at fstadv.R#157
7.
get_fstadv(url) at data-quality.R#33
6.
.f(.x[[i]], ...) 
5.
map(.x = nums, .f = function(x) {
    url <- storms %>% slice(x) %>% .$Link
    key <- storms %>% slice(x) %>% .$Key
    obs.fstadv <- get_fstadv(url) ... at data-quality.R#30
4.
eval(ei, envir) 
3.
eval(ei, envir) 
2.
withVisible(eval(ei, envir)) 
1.
source("~/Projects/Hurricanes/data-raw/data-quality.R") 
```

**Comment by timtrice** [comment]
Error is occurring in `fstadv_extract_forecasts`. In [Adv 6 for EP031999](http://www.nhc.noaa.gov/archive/1999/mar/MEP0399.006.html), forecast positions are:

```
REPEAT...CENTER LOCATED NEAR 14.6N 114.8W AT 15/1500Z 
AT 15/1200Z CENTER WAS LOCATED NEAR 14.8N 114.2W 
  
FORECAST VALID 15/2100Z...DISSIPATED 
  
REQUEST FOR 3 HOURLY SHIP REPORTS WITHIN 300 MILES OF 14.6N 114.8W 
```

The "DISSIPATED" text likely throwing off the regex.

**Comment by timtrice** [comment]
Appended conditional statement when checking for forecasts in `fstadv_forecasts`:

```
    # If no forecasts, exit gracefully
    if (length(ds) == 0 | purrr::is_empty(ds[[1]]))
        return(NULL)
```

**Comment by timtrice** [comment]
Originally, fstadv_forecasts checked for length of ds and returned
NULL if 0. However, the incomplete forecast position made ds a
length 1 which would pass but generate the error on further
conversion.

Now, we check for all list elements of ds. If all elements check
TRUE on purrr::is_empty, then return NULL; otherwise, proceed.

Issue resolved per commit b37f354.
